internal/query/relations.go

```diff
@@ -631,13 +834,35 @@ func (e *engine) ListRoles(ctx context.Context, resource types.Resource) ([]type
 		},
 	}
 
+	dbRoles, err := e.store.ListResourceRoles(ctx, resource.ID)
+	if err != nil {
+		return nil, err
+	}
+
 	relationships, err := e.readRelationships(ctx, filter)
 	if err != nil {
 		return nil, err
 	}
 
 	out := relationshipsToRoles(relationships)
 
+	rolesByID := make(map[string]storage.Role, len(dbRoles))
```

Comment:
Maybe a silly question, but why not just have a `map[gidx.PrefixedID]storage.Role` if we just convert every key to a string?

Reply:
🤔 that's a good question... i'm not sure why I did that... It definitely doesn't need to be converted. Thanks!